### PR TITLE
vasprintf fails by returning -1

### PR DESCRIPTION
**Pull request by krocard**
int vasprintf(char **strp, const char *fmt, va_list ap)
return -1 on failure, not by setting strp to NULL.

Replace check for NULL by checking for -1 return.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/01org/parameter-framework/pull/27%23issuecomment-65069426%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/27%23issuecomment-65069551%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/27%23issuecomment-65228778%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/27%23issuecomment-77661429%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/27%23issuecomment-89396447%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/27%23issuecomment-90470592%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/27%23issuecomment-90475919%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/27%23issuecomment-132997983%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/27%23discussion_r37765174%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/27%23issuecomment-134255939%22%2C%20%22https%3A//github.com/01org/parameter-framework/pull/27%23issuecomment-135367862%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/27%23issuecomment-65069426%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%2B1%22%2C%20%22created_at%22%3A%20%222014-12-01T14%3A13%3A35Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6605131%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/gdenneul%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222014-12-01T14%3A14%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%20%22%2C%20%22created_at%22%3A%20%222014-12-02T13%3A16%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5653633%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Makohoek%22%7D%7D%2C%20%7B%22body%22%3A%20%22Commits%20were%20rebased%20without%20modification%20due%20to%20github%20refusing%20to%20merge%20them%20otherwise.%22%2C%20%22created_at%22%3A%20%222015-03-07T00%3A31%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%40dawagner%20Can%20you%20review%20%3F%20This%20pull%20request%20is%204%20month%20old%20now%20%21%22%2C%20%22created_at%22%3A%20%222015-04-03T19%3A22%3A26Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22The%20varargs%20are%20going%20away%20thanks%20to%20%40clero%20%27s%20pull%20request%20anyway.%22%2C%20%22created_at%22%3A%20%222015-04-07T08%3A55%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6430928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/dawagner%22%7D%7D%2C%20%7B%22body%22%3A%20%22I%20guess%20you%20are%20talking%20about%20%2358.%20I%20agree%20that%20this%20pull%20request%20is%20far%20better.%20Nevertheless%20I%20hope%20it%20will%20be%20merged%20one%20day.%20Nobody%20reviewed%20it%20yet%20and%20their%20is%20still%20opened%20questions%20about%20it%20%28especially%20the%20commits%20delegating%20log%20to%20clients%29.%20Plus%20this%20pull%20request%20is%20rather%20small%20and%20easy%20to%20rebased%20on.%22%2C%20%22created_at%22%3A%20%222015-04-07T09%3A07%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22Reopen%20because%20%2358%20was%20never%20merged%20in%20master%20and%20the%20rebase%20will%20not%20be%20that%20difficult%20if%20it%20is%20merged%20%28just%20delete%20the%20function%29.%22%2C%20%22created_at%22%3A%20%222015-08-20T12%3A52%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A-1%3A%5Cr%5Cn%22%2C%20%22created_at%22%3A%20%222015-08-24T15%3A42%3A09Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%2C%20%7B%22body%22%3A%20%22Solved%20by%20log%20rework%20in%20%23168%20will%20be%20merged%20in%20windows_port%20then%20master%20shortly.%22%2C%20%22created_at%22%3A%20%222015-08-27T09%3A51%3A44Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6862950%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/krocard%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20b7bfb4b581150fa09189ddc5de008e712950da38%20parameter/Element.cpp%2048%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/01org/parameter-framework/pull/27%23discussion_r37765174%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22coding%20style%22%2C%20%22created_at%22%3A%20%222015-08-24T15%3A39%3A06Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/820313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/OznOg%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20parameter/Element.cpp%3AL79-97%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/27#issuecomment-65069426'>General Comment</a></b>
- <a href='https://github.com/gdenneul'></a> +1
- <a href='https://github.com/krocard'></a> :+1:
- <a href='https://github.com/Makohoek'></a> :+1:
- <a href='https://github.com/krocard'></a> Commits were rebased without modification due to github refusing to merge them otherwise.
- <a href='https://github.com/krocard'></a> @dawagner Can you review ? This pull request is 4 month old now !
- <a href='https://github.com/dawagner'></a> The varargs are going away thanks to @clero 's pull request anyway.
- <a href='https://github.com/krocard'></a> I guess you are talking about #58. I agree that this pull request is far better. Nevertheless I hope it will be merged one day. Nobody reviewed it yet and their is still opened questions about it (especially the commits delegating log to clients). Plus this pull request is rather small and easy to rebased on.
- <a href='https://github.com/krocard'></a> Reopen because #58 was never merged in master and the rebase will not be that difficult if it is merged (just delete the function).
- <a href='https://github.com/OznOg'></a> :-1:
- <a href='https://github.com/krocard'></a> Solved by log rework in #168 will be merged in windows_port then master shortly.
- [ ] <a href='#crh-comment-Pull b7bfb4b581150fa09189ddc5de008e712950da38 parameter/Element.cpp 48'></a> &nbsp;<b><a href='https://github.com/01org/parameter-framework/pull/27#discussion_r37765174'>File: parameter/Element.cpp:L79-97</a></b>
- <a href='https://github.com/OznOg'></a> coding style

<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/27?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/01org/parameter-framework/pull/27?approve=1'></a>&nbsp;<a href='https://github.com/01org/parameter-framework/pull/27'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by krocard** [comment]
Commits were rebased without modification due to github refusing to merge them otherwise.

**Comment by krocard** [comment]
@dawagner Can you review ? This pull request is 4 month old now !

**Comment by dawagner** [comment]
The varargs are going away thanks to @clero 's pull request anyway.

**Comment by krocard** [comment]
I guess you are talking about #58. I agree that this pull request is far better. Nevertheless I hope it will be merged one day. Nobody reviewed it yet and their is still opened questions about it (especially the commits delegating log to clients). Plus this pull request is rather small and easy to rebased on.

**Comment by krocard** [comment]
Reopen because #58 was never merged in master and the rebase will not be that difficult if it is merged (just delete the function).

File: parameter/Element.cpp [hunk 1]
```diff
@@ -99,6 +79,18 @@ void CElement::log_table(bool bIsWarning, const std::list<string> lstrMessage) c
     }
 }
 
+void CElement::vLog(bool bIsWarning, const char *strMessage, va_list args) const
+{
+    char *pacBuffer;
+    if(vasprintf(&pacBuffer,  strMessage, args) == -1) {
```
Comment (OznOg) [inline parameter/Element.cpp]: coding style

**Comment by OznOg** [comment]
:-1:

**Comment by krocard** [comment]
Solved by log rework in #168 will be merged in windows_port then master shortly.

**Pull request closed without merging**